### DISCUSSION TITLE
Communication, Transparency, and Meeting Minutes

### DISCUSSION THREAD

**Discussion by Azure3141** [endless-sky/endless-sky#10421] [General]
I will preface this by saying it was largely inspired by #10412, specifically the section on hidden channels and discussions.

> There is absolutely no need to have hidden channels on the discord server, it should be common knowledge what the main contributors talk about. Acceptable restrictions might be write permissions or the moderation room.

This is an issue that is subtle, and which probably goes unnoticed by the vast majority of both players and contributors, but certain parts of it have an immense impact on the relationship between contributors and the developers. 

## Hidden Channels
First, which channels are hidden, and why? As far as I am aware, there are six permission-restricted channels on the ESC discord (ignoring the opt-in offtopic channels):

<details>
  <summary>Private Channels</summary>
For Core Members:
#inner-circle
- Hangout and drama channel

#high-class-content
- Essentially a private #content-creating
- As I understand it, created after an incident where unfinished content made by a core member was shared publicly and then stolen
- Rarely used these days. I've only really ever seen it used when devs/core members wish to check mission syntax or ask how to do something complex with a mission. Truthfully, nothing about that is secret and my guess is that they end up posting in #high-class-content mostly so that their questions aren't buried in the endless scrawl of #content-creating
- You occasionally have someone showcasing a spoiler from an upcoming storyline, but this is rare

For Reviewers:
#reviewer-conclave
 - A channel where reviewers discuss the status of PRs they are reviewing. Pretty inactive and has been described as "irrelevant" because of how unused it is.

For Moderators / Community Organizers
#moderator-chat (presumptive name)
- Presumably moderation actions are discussed here?

#community-organizer-chat (presumptive name)
- Truthfully, not sure this channel is used much at all since community manners usually come up in #inner-circle and we only have one active Community Organizer anyway

For Developers
#developer-chat (presumptive name)
- This is the focus of this discussion. The public **perception** is at least that this is where the developers discuss plans and decisions being made.
- This channel is also paired with a private developer voice chat, where the developers conduct weekly meetings, where plans and decisions are known to be made.
</details>

The hidden status of the moderation and community organization chats is probably not a big issue. Some moderation discussions are sensitive and it makes sense for them to be hidden, while it doesn't seem like the community organizer chat is used much, if at all. In addition to this, the reviewer channel is pretty much unused outside some coding discussions. I would perhaps make #reviewer-conclave read-only for non reviewers, allowing others to see the discussions in it, but I don't think it's a huge concern.

## Two Barriers to Access

With that out of the way, we can talk about there being essentially two levels of separation with the remaining channels: Between the general community and the Core Members, and between the general community / core members and the Developers. In the interest of fairness, I will discuss both here, though the proposal outlined here is more geared towards addressing the latter barrier. However, to get to the actionable solutions I would like to propose, you can skip the following spoiler for brevity.

<details>
  <summary>Core Members</summary>

So, #inner-circle and #high-class-content. These mark the barrier between the general community and Core Members. In all honesty, #inner-circle *does* function at least somewhat as a semi-exclusive club (with new Core Members added via vote, drawn from the active content creators and contributors) for the upper strata of the community. It's pretty common as the go-to place for casual conversation that might not fit with a topic channel elsewhere on the server, while not being quite as chaotic as the actual #off-topic channel. It also sees use as a discussion channel for various levels of community drama, such as complaints that people might not be comfortable airing in public, but still wish to be seen by a smaller group of people that they interact with regularly. Now, is this a problem? I'd say it could be construed as one under the lines of egalitarianism-- but also, it's a discord server. A semi-private hangout room for the most active members is honestly not a massive injustice, and isn't what I want to focus on here.

As for #high-class-content, issues have been raised with it in the past, and I largely believe those issues were more relevant to how the channel was used at the time. I can definitely understand the concern with there being a stratification of content creators-- if all core members only posted their content in #high-class-content, that would be detrimental to activity in the rest of the server, especially with regards to helping newer people learn and start making content of their own. However, in my experience over the last few years, this isn't how the channel is used. For one, there is activity in the channel only a couple times a month at most, and most of the time the activity is just someone asking technical questions about the mission syntax or how to do something complex. I believe this is posted to #high-class-content largely because of how fast the chat moves in #content-creation, so posting a detailed question like that in a more exclusive channel where the more experienced mission creators are likely to see it is a better way to ensure your question is actually answered. I do think that it would be ideal if these posts could be seen in public, though, such that people could learn from the explanations given in the channel. To this end, the solution I propose here is to, **as with the project channels, make #high-class-content visible to everyone, but only give write permissions to Core Members.** The channel was originally created as a private one out of concerns that unfinished, showcased content might be stolen, but honestly I think that is a fringe concern in the modern ES community and shouldn't define the purpose of an entire channel. A read-only solution would both preserve the current as-used purpose of the channel for ensuring questions aren't buried, while also keeping things open such that newer content creators are still able to see what goes on.

</details>

There is likewise a barrier between the general community / Core Members, and the Developer team. Perhaps it is out of bias, but I believe this much more strongly impacts Core Members than the general community, if only because Core Members are usually the ones making content for the game-- and content creation is usually where this barrier rears its head. The Campfire Chats, which are effectively Q&As between the developers and the general community, in my opinion adequately address the questions a casual player might have about the development direction of the game. However, they fall short when it comes to being a good way for more active content creators to understand where the Developers are on a particular plan or decision, due to their infrequency and to how quickly decisions can change.

To this end, occasional "Core Cookouts," essentially Campfire Chat style Q&As between the Developers and whatever Core Members show up, have also been happening monthly (albeit infrequently, we've missed a couple in recent months). These have the issues of being unannounced (unless you know the schedule) and taking place at times not everyone is free to meet. In addition to this, my experience with the Core Cookouts is that the Developers dislike talking about an issue in detail, and there is a constant push to "move on" to something else, even when there are few other issues to discuss, which usually results in the Core Cookout ending. This is a suboptimal format when it comes to my concerns with how communication between Core Members and Developers is conducted.

Plans and decisions are usually made by the Developers during their weekly meetings, which are conducted in private. These often, if not always, involve discussing the work of content creators and when/if they will be reviewed or merged. Occasionally, in the past a content creator has been invited to discuss their content with the dev team, but this has happened only a tiny handful of times in the past few years, whereas plans for people's work are discussed outside of their knowledge weekly. This is where the Core Cookouts explicitly fail, because they take a Q&A format where content creators have to essentially ask "is anything happening with my content?" instead of being informed about those decisions in the first place, which again happens frequently, with equally frequent changes.

So, we have a situation where every week, the Developers talk about their plans and make decisions in a private voice chat, often about other people's work, and those people are left completely in the dark about what these decisions exactly are. This is a persistent problem that repeatedly re-emerges, even after promises of better communication that have universally fallen short.

I have created this as a separate post to the aforementioned discussion because I want to propose an actionable item that could be taken by the developers to resolve this issue. This is something I believe could be implemented at the very next Developer meeting, with immediate results, and which I urge the Developers to at minimum use on a trial basis for the next few weeks.

## A Solution
As you might expect, the problem of communicating major decisions made in high-level meetings often comes up in industry and professional settings, and a solution has long since been devised. Essentially, the Developers, or someone else present in their meetings (a Secretary role) should take meeting minutes (aka notes) during their weekly meetings, in which **the agenda, points discussed, decisions reached, and overall plans / changes to plans** are recorded and released for viewing. To be frank, meeting minutes should be an expectation for an organization of this scale, both for communication and accountability issues.

While I understand this is a volunteer organization, and the burden of taking notes during a voice chat might add extra work for whichever Developer or additional person is relegated to taking them. However, when discussions and decisions are being made about the work of other people, especially volunteers, I believe there is an obligation to record the decisions made about them and communicate them to the people with a stake in them. As it cannot be relied on for the Developers to consistently remember to update people without prompting, I think a consistent system where meeting notes are taken and released would lead to a better communication situation. Some might say this is an over-professionalization of ES as a volunteer game, but I believe if we are going to have a hierarchy of Developers as decision makers, keeping to a professional standard of record-keeping and communication is something that should be held to as well.

In summary, the Developers should take meeting minutes, following a readable format and adequately recording:
1. The meeting's agenda, or what was discussed
2. Important points in each discussion
3. Decisions that were made during these discussions
4. What these decisions change about the overall plans for development (which should likewise be recorded)

This should be as detailed as is necessary to convey enough information that a content creator might read these notes, recognize that something they have a stake in was mentioned, and either be able to fully understand what was said in relevant discussion or have enough information to reach out and ask the appropriate questions to complete their understanding.

These meetings should, at minimum, be released to Core Members (as they are usually the content creators with the most relevant stake in this issue). However, as there are plenty of prolific content creators who are *not* Core Members, it is my view that they should be available for public viewing by everyone. If this is not desirable, (say, due to wanting to not overpromise for a future update) then someone on the Development team should keep careful note of which non-Core Members had their work discussed during the meeting and DM them the notes separately.

tl;dr devs should take notes during their meetings.

**Comment by mOctave** [comment]
I think this is a very accurate and useful summary of the situation, and fully agree with you regarding all the points you've made*.

I also have a couple more points to note.

Firstly, I think that "important points" is a bit of a vague term which could end up going any one of a number of ways. I think it's important that these important points include the reasoning behind any decisions. Right now, the number one complaint that content creators are having, as far as I'm aware, is that the devs are making poorly informed decisions about their work without the content creators knowing the reasons why the decisions were made. Having a public list of reasons why their work was rejected or modified will either restore content creators' faith in the devs, or, in the worst case, give them specific points to argue against the devs' decisions. To be fair, I think this is something that's being done fairly well already when it comes to closing PRs, but the issue still exists with PRs that are replaced or modified, or with ideas that haven't gotten to the PR stage yet.

I also believe that these minutes should not be released via Discord, or at least, not only via Discord. I think having a public, announced GitHub repository or Google Drive folder would be the best solution, because although most content creators are on Discord, not all are (and some have intentionally left the ESC server). On the other hand, just about every content creator can access GitHub or Google Drive.

If the devs don't want public releases, then I would suggest the same format, but manually adding content creators. That would be more work, though, and would leave out everyone who was following along with something without actively contributing to it.

*With the one exception being that I can still see situations in which a plugin creator, for example, might want help on something in #high-class-content without leaking spoilers to the wider community, so I'm still on the fence about keeping that channel private... although that could also happen in #the-inner-circle, so it might be fine.

**Reply by Azure3141** [reply]
I agree on what you said about important points. I left it a bit flexible for the sake of the devs, but including the reasoning behind decisions would go a long ways towards providing insight into those decisions.

Ideally, posting these notes on Github would work well. However, I think there might be some pushback against publicly posting these due to concerns that things might be discussed in a dev meeting (e.g. the new goal for XYZ update is the Republic campaign) and then taken as promises by the general community, which might not work out. If they can be posted on Github but only viewable to a smaller group of people, that would work but might make it more difficult to set everything up in terms of permissions.

To be honest, I don't think the concern about spoilers in ES content creation is an important one. The files are all easily openable with any text editor, so the barrier to spoiling yourself is already so low that I don't think we should be worried about people reading through unfinished, WIP mission files. Of course, it might be different for *releasing* new plugins or storylines whose existence itself is wanted to be kept as a surprise, but in that case asking the cast of Core Members for assistance is basically revealing it to a large fraction of the active server population anyway.

**Reply by mOctave** [reply]
> Ideally, posting these notes on Github would work well. However, I think there might be some pushback against publicly posting these due to concerns that things might be discussed in a dev meeting (e.g. the new goal for XYZ update is the Republic campaign) and then taken as promises by the general community, which might not work out. If they can be posted on Github but only viewable to a smaller group of people, that would work but might make it more difficult to set everything up in terms of permissions.

There could always just be a big notice on the README saying "This is a record of the private discussions of the developers of Endless Sky. Nothing in these minutes constitutes an announcement promise regarding what changes will or won't be made to the game. The minutes are made public solely for the sake of transparency."

> Of course, it might be different for releasing new plugins or storylines whose existence itself is wanted to be kept as a surprise, but in that case asking the cast of Core Members for assistance is basically revealing it to a large fraction of the active server population anyway.

This is what I was referring to. I was thinking that, although most active and experienced content creators are Core Members, most active players still _aren't,_ so while a Core Member might think "oh this looks like a cool plugin X is working on," most of the potential audience would still be unaware until it got released.

Having done a bit more thinking, though, this could probably happen in #the-inner-circle, and so I think you're probably right that #high-class-content should be public to read.

**Reply by Azure3141** [reply]
A README on the notes would work as something the devs could point at after the fact, but I'm doubtful of how much that would actually affect people's feelings about perceived "promises" or "abandoned projects," especially around things as sensitive as the Republic and Syndicate campaigns.

As for #high-class-content, a solution to that could possibly be making it an opt-in channel? So it's still public for those who know what they're getting into, but a casual player browsing through the channels won't accidentally stumble upon spoilers.

**Reply by mOctave** [reply]
> A README on the notes would work as something the devs could point at after the fact, but I'm doubtful of how much that would actually affect people's feelings about perceived "promises" or "abandoned projects," especially around things as sensitive as the Republic and Syndicate campaigns.

Maybe. But if it was established with that explicitly stated, I don't think it would be taken as promises, especially since it would mostly be senior content creators who read the minutes. At least, it wouldn't be taken as a promise any more than the casual comments/jokes of the devs on Discord already are.

> As for #high-class-content, a solution to that could possibly be making it an opt-in channel? So it's still public for those who know what they're getting into, but a casual player browsing through the channels won't accidentally stumble upon spoilers.

Sounds like a good idea.

**Comment by reticent-rem** [comment]
Inspired by the campfire chatroom today, I have a more lightweight potential solution that could be an improvement for helping content creators understand where the Developers are on a particular plan or decision, and help the Developers keep track of what the community feels is important on a more tactical level. 

Would having a weekly google form focused on priorities for the week, somewhat like the campfire chat submissions, be helpful without adding too much administrative burden to Developers? And then the questions & answers can be posted for the community somewhere (could be a public google doc, could be a discord information room post, whatever method is easiest & most convenient for Developers to start and then go from there).

I envision the intake being something like:
"This intake is focused on short term questions to Developers on priorities or questions about the upcoming week"

1. What is your question? 
2. Is this related to an existing Issue or PR? If so, which one?

That way, instead of a content creator pinging a Developer on discord or bumping a PR when things sometimes are forgotten or missed, there's a structured, centralized place to ask Developers questions and get answers during or after the weekly Developer meeting, and the Developers get some agenda items directly from the community. Walking through a few potential examples:

- "My PR is ready for Developer review. Will it be reviewed this week?" 
  - "Yes, we'll try to review abc part this week." The Developers are reminded about this PR and can make it a priority to review.
  - "No, not likely to be reviewed this week or next week. It looks like there are still some unresolved comments from xyz." The content creator learns that the Developers are still waiting for some work to be done before review, and get transparency that it won't get looked at for 2 weeks. They can make edits and submit a follow-up form in 2 weeks, or otherwise address the Developer response.

- "Is work being done on issue xyz?"
  - "Nope" Community gets clarification on an issue being worked on or not by devs.

- "I'm thinking of making a PR that does abc as outlined in issue xyz. Is this something that could be worked into vanilla if I make a PR, or is it likely to be vetoed by Developers?"
  - "Feel free to make a PR, the general concept as a whole seems reasonable." The Developers learn that someone's planning to resolve an issue and get early input & insight of things in the content pipeline.
  - "No, this doesn't work because blah blah blah" The Developers can step in early in the process so that there isn't a ton of wasted work by content creators.

- "Incoherent, very dramatic, or off topic question here"
  - "Not addressed." Developers don't need to address every question, but having that transparency is important and if others believe the question is important it can be brought up in other forums.

Broadly I think this can address a lot of the problems that content creators have with missing Developer feedback, helps Developers with keeping track of tactical things important to community members during their weekly meetings, and doesn't add too much administrative overhead to a group of volunteers. Or at the very least, hopefully be a step in the right direction.

**Reply by Azure3141** [reply]
I think the problem isn't with the ability for people to ask questions of the developers, as pinging them or DMing them is pretty easy. The issue with this is that unless you're asking essentially the same question ("did my PR come up in discussions?") every week, you just won't be informed when those decisions are made. This is what meeting notes are meant to alleviate, informing content creators when something relevant to them comes up rather than expecting them to be constantly asking about it or hoping a developer decides to reach out to them on their own, which happens only when a dev *thinks* (and remembers) someone should have that information, which rarely happens as often as I'd like in my experience.

Adding another form for the devs to review every week seems like a lot more work compared to just having someone write brief notes (they can just be bullet points) during their meetings. Despite what was said about it being too much work, taking notes of a meeting is a very small ask and is something I believe is a reasonable requirement for any semi-professional organization.

**Reply by warp-core** [reply]
> The issue with this is that unless you're asking essentially the same question ("did my PR come up in discussions?") every week, you just won't be informed when those decisions are made.

I don't really understand this. What would happen if there was a discussion about something and then none of what was discussed or concluded is shared? It seems to me that whatever was discussed is essentially jsut discarded. If we feel something should be changed, but that idea is never shared, it's unlikely that whatever is desired will happen.

**Reply by Azure3141** [reply]
> I don't really understand this. What would happen if there was a discussion about something and then none of what was discussed or concluded is shared? It seems to me that whatever was discussed is essentially jsut discarded. If we feel something should be changed, but that idea is never shared, it's unlikely that whatever is desired will happen.

Because this still affects content creators when you make decisions, even if it isn't shared. Those decisions aren't discarded, you just don't inform content creators about them unless they go out of their way to ask. There have been numerous instances where I have only found out a decision was made by asking about the status of a PR or project, because the developers either forgot or didn't bother to inform me.

For example, I was told by a developer that the plan was to get the Avgi into the upcoming update. This caused me to reprioritize tasks, both with ES projects and IRL plans, to focus on the Avgi and ensure they were ready for merge. Later I found out through an offhand comment that a decision had been made to not include the Avgi in the upcoming update, which had happened a week prior but which no one had told me about. Had this change been documented in meeting notes, or had someone simply bothered to tell me about it, this would not have been an issue.

**Comment by Tangle10** [comment]
I'm not sure if it's permission-restricted, but also the Republic Campaign Dev channel that I'm pretty sure exists doesn't show up while all the other community project rooms do.

**Reply by Saugia** [reply]
If a channel is not visible, it means there's nothing to be visible, no conversations, decisions, any of the likes going on with it. Even if we did have a channel for it, there would be nothing in it, and once something was in it or prepared progress, it would be visible anyways.

**Reply by Tangle10** [reply]
> If a channel is not visible, it means there's nothing to be visible, no conversations, decisions, any of the likes going on with it. Even if we did have a channel for it, there would be nothing in it, and once something was in it or prepared progress, it would be visible anyways.

Fair enough! I had figured it must exist and that it being invisible would probably just be a mistake that the discord's admins wouldn't be aware of because. Well. How would anyone know without seeing it, but it's nice to know for sure (another reason why one might chafe at invisible channels...)
